### PR TITLE
beef up --resolv-conf= options of systemd-nspawn

### PR DESCRIPTION
**Pull request by poettering**
An attempt to improve the #15340 situation

### CODE REVIEW

File: man/systemd-nspawn.xml [hunk 1]
```diff
@@ -1099,29 +1099,60 @@
...
+        <para>If set to <literal>copy-host</literal>, the <filename>/etc/resolv.conf</filename> file from the
+        host is copied into the container, unless the file exists already and is not a regular file (e.g. a
+        symlink). Similar, if <literal>replace-host</literal> is used the file is copied, replacing any
+        existing inode, including symlinks. Similar, if <literal>bind-host</literal> is used, the file is
+        bind mounted from the host into the container.</para>
+
+        <para>If set to <literal>copy-static</literal>, <literal>replace-static</literal> or
+        <literal>bind-static</literal> the static <filename>resolv.conf</filename> file supplied with
+        <citerefentry><refentrytitle>systemd-resolved.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+        (specifically: <filename>/usr/lib/resolv.conf</filename>) is copied or bind mounted into the
```
Comment (keszybz) [inline man/systemd-nspawn.xml]: /usr/lib/systemd/resolv.conf

**Comment by poettering** [comment]
> With your patch, we only have 12. But the missing ones, like bind but with "replace", are also useful. The original report was about a case where the user wants to force a certain configuration from outside of the container, without caring how exactly /etc/resolv.conf is set up in the container. And they might want to use "bind" too. So I think this should be changed to [force-]{copy,bind}-{host,static,uplink,stub}.

Hmm, so my assumption was that "bind" only is useful if the image is read-only, or if you don't want to modify the image at all. But in either case you can't have or don't want use to replace anything. i.e. why would anyone want to use bind mounting when removing what might be underneath? I simply don't see the usecase for that?

I thought about doing a "force" flag first, but I then realized that coming from the usecase PoV the "force bind" situation doesn't really make sense.

I mean, bind mounting the file kinda sucks, because people might think it would mean continious propagation of the host settings into the container. But htat's not what happens, since /etc/resolv.conf is generally updated atomically, i.e. gets replaced by a new inode, thus disconnecting the bind mount. Given that things are that way it almost never makes sense to use bind mounts — except the image you work with really can't or shouldn't be modified. 

does that make sense?

**Comment by poettering** [comment]
force pushed a new version. only change is the resolv.conf path correction. i left the "replace" verb as is, see discussion. ptal

**Comment by keszybz** [comment]
You're right. I didn't think that the bind mounts don't actually keep them synchronized.

**Comment by keszybz** [comment]
bionic-i386 timed out in upstream tests as it often does.

**Pull request merged**